.github/workflows/build-release.yml

```diff
@@ -6,44 +6,58 @@ on:
 
 jobs:
   build:
-
-    runs-on: ${{ matrix.os }}
+    name: ${{ matrix.spec.name }}
+    runs-on: ${{ matrix.spec.runner }}
     env:
       BUILD_NUMBER: ${{ github.run_number }}
     strategy:
+      fail-fast: false
       matrix:
-        os:
-          - ubuntu-18.04
-          - macOS-10.15
-          - windows-latest
+        spec:
+        - { name: 'Linux x86_64', runner: 'ubuntu-18.04' }
+        - { name: 'Linux ARM', runner: 'ubuntu-18.04', toolchain: 'Linux-arm', no_tests: 'true' }
+        - { name: 'MacOS x86_64', runner: 'macOS-10.15' }
+        - { name: 'MacOS arm64', runner: 'macOS-11', toolchain: 'macOS-arm64', no_tests: 'true' }
+        - { name: 'Windows x86_64', runner: 'windows-2019' }
 
     steps:
       - uses: actions/checkout@v2
         with:
           fetch-depth: 0
+
+      - name: Install Crossbuild tools
+        if: matrix.spec.runner == 'ubuntu-18.04'
+        run: |
+          sudo apt update -y
+          sudo apt install -y crossbuild-essential-armhf crossbuild-essential-arm64
+
       - name: setup/build CMake
+        env:
+          TOOLCHAIN: ${{ matrix.spec.toolchain && format('{0}/toolchains/{1}.cmake', github.workspace, matrix.spec.toolchain) || '' }}
         run: |
           echo "build number = ${BUILD_NUMBER}"
           cmake -E make_directory ${{runner.workspace}}/build
-          cmake -S ${{ github.workspace }} -B ${{runner.workspace}}/build
+          cmake -DCMAKE_TOOLCHAIN_FILE="${TOOLCHAIN}" -S ${{ github.workspace }} -B ${{runner.workspace}}/build
           cmake --build ${{runner.workspace}}/build
+
       - name: make artifacts
-        run: cmake --build ${{runner.workspace}}/build --target ziti-prox-c --target package
-      - name: zip artifacts
-        if: runner.os != 'Windows'
-        run: 7z a -tzip ziti-prox-c-${{ runner.os }}.zip ${{runner.workspace}}/build/programs/ziti-prox-c/ziti-prox-c
-      - name: zip artifacts (WIN)
-        if: runner.os == 'Windows'
-        run: 7z a -tzip ziti-prox-c-${{ runner.os }}.zip ${{runner.workspace}}/build/programs/ziti-prox-c/Debug/ziti-prox-c.exe
-      - name: Get release
-        id: get_release
-        uses: bruceadams/get-release@v1.2.2
-        env:
-          GITHUB_TOKEN: ${{ github.token }}
+        run: cmake --build ${{runner.workspace}}/build --target ziti-prox-c-pack --target package
+
+#      - name: zip artifacts
+#        if: runner.os != 'Windows'
```

Comment:
Could we use `cmake -E tar --format=zip` on all operating systems?

Reply:
not necessarily, support for `--format=zip` is not guaranteed.
In practice it is only used by our CI